validator/sawtooth_validator/journal/block_store.py

```diff
@@ -56,10 +56,10 @@ def __contains__(self, x):
         return x in self._block_store
 
     def __iter__(self):
-        return iter(self._block_store)
+        raise NotImplementedError('BlockStore is not iterable')
```

Comment:
Perhaps a comment here on this being non-trivial, and only here as a requirement of the abstract base class.